streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -54,14 +55,17 @@
     private JoinWindows(String name, long before, long after) {
         super(name);
 
-        if (before < 0) {
-            throw new IllegalArgumentException("window size must be > 0 (you provided before as " + before + ")");
+        if (before < 0) { // shift lower bound to right
```

Comment:
Can we just simply check `after + check >= 0`, and if failed throw message that `interval length cannot be negative`?